pkg/apis/application/v1alpha1/types.go

```diff
@@ -2295,6 +2295,17 @@ func (dest ApplicationDestination) Equals(other ApplicationDestination) bool {
 	return reflect.DeepEqual(dest, other)
 }
 
+// Equals compares two instances of ApplicationSpec and returns true if instances are equal.
+func (spec ApplicationSpec) Equals(other ApplicationSpec) bool {
```

Comment:
Can you add unit tests to demonstrate the issue and prove the solution?

We might also want to make a note on the ApplicationSpec struct that whenever a field is added, you. gotta update this function.